src/Microsoft.NetCore.Analyzers/Core/Runtime/AvoidZeroLengthArrayAllocations.Fixer.cs

```diff
@@ -46,20 +48,50 @@ public sealed override FixAllProvider GetFixAllProvider()
             SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             SyntaxGenerator generator = editor.Generator;
 
-            SyntaxNode variableDecl = generator.GetDeclaration(nodeToFix, DeclarationKind.Variable);
-            SyntaxNode typeNode = generator.GetType(variableDecl);
+            var operation = semanticModel.GetOperationInternal(nodeToFix, cancellationToken);
```

Comment:
Why is this calling `GetOperationInternal` instead of `GetOperation`?

Reply:
I saw some other code that used the `Internal` overloads of the IOperation APIs, so I followed. e.g. There are a lot of `RegisterOperationActionInternal` calls.

Reply:
Interesting, I thought I recently removed all the Internal APIs and calls. Did you sync your sources with latest master?

Reply:
It should have been deleted with https://github.com/dotnet/roslyn-analyzers/pull/1334/files#diff-5cbfbc6801faa8ac7bc5ae144689bb70L155. Please resync to latest.

Reply:
A simpler way to implement this would be to use [FindNode](http://source.roslyn.io/#Microsoft.CodeAnalysis/Syntax/SyntaxNode.cs,1973a2affa06beb8,references) API for the diagnostic's location span, and then request the type associated with the array creation using `SemanticModel.GetTypeInfo(node)` and check if it is an array type. This will avoid you the trouble of having to figure out what operation node maps to the diagnostic span.